flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/module/hive/HiveModuleTest.java

```diff
@@ -88,4 +94,24 @@ public void testHiveBuiltInFunction() {
 	public void testNonExistFunction() {
 		assertFalse(new HiveModule(HiveShimLoader.getHiveVersion()).getFunctionDefinition("nonexist").isPresent());
 	}
+
+	@Test
+	public void testConstantArguments() throws Exception {
+		TableEnvironment tEnv = HiveTestUtils.createTableEnvWithBlinkPlannerBatchMode();
+
+		tEnv.unloadModule("core");
+		tEnv.loadModule("hive", new HiveModule(HiveShimLoader.getHiveVersion()));
+
+		List<Row> results = TableUtils.collectToList(tEnv.sqlQuery("select concat('an', 'bn')"));
+		assertEquals("[anbn]", results.toString());
+
+		results = TableUtils.collectToList(tEnv.sqlQuery("select concat('ab', cast('cdefghi' as varchar(5)))"));
```

Comment:
Can you add a test for `null` literal?

Reply:
Unfortunately we cannot test `null` literals, because Hive functions consider nulls as non-constant arguments: https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/HiveFunction.java#L36

Maybe we should consider all the passed arguments as constant, and don't pass non-constant arguments in the first place. But that'll need to change the interface.

Reply:
OK.
Is there any cases to cover null? If not, we can add some whatever whether should be a literal.

Reply:
And you can add a TODO to the tests.
About null literal support, we should wait for UDF type reworking.